wonnx/templates/endomorphism/gather.wgsl

```diff
@@ -0,0 +1,33 @@
+{%- include "structs.wgsl" -%}
+
+struct Indices {
+	data: [[stride(4)]] array<i32>;
+};
+
+struct Chunk {
+	data: [[stride({{ scalar_stride * chunk_size }})]] array<{{ chunk_type }}>;
+};
+
+[[group(0), binding(0)]]
+var<storage, read> input_0: Chunk; // data
+
+[[group(0), binding(1)]]
+var<storage, read> input_1: Indices; // indices
+
+[[group(0), binding(2)]]
+var<storage, write> output_0: Chunk;
+
+[[stage(compute), workgroup_size({{ workgroup_size_x }}, {{ workgroup_size_y }})]]
+fn main([[builtin(global_invocation_id)]] global_id: vec3<u32>) {
+	let index_index = global_id.x; // Index of the index in the indices array that we are currently processing
+	let chunk_index = global_id.y; // Chunk of elements that we are copying for this index (chunk size determined dynamically)
+	let index_stride = u32({{ i_chunks[0][0] / chunk_size }});
```

Comment:
Wouldn' t that be slightly more costly than:
`let index_stride = {{ i_chunks[0][0] / chunk_size }}u;`

Reply:
Well, a smart compiler should be able to figure out that both expressions resolve to a constant, not sure how smart this compiler is :-) I will change to your suggestion.